### PR TITLE
kyotocabinet: New package

### PR DESCRIPTION
**Pull request by EbiArnie**
This adds Kyoto Cabinet, an in-process database manager

### CODE REVIEW

File: var/spack/repos/builtin/packages/kyotocabinet/package.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    """Kyoto Cabinet is a library of routines for managing a database."""
+
+    homepage = "https://dbmx.net/kyotocabinet/"
+    url = "https://dbmx.net/kyotocabinet/pkg/kyotocabinet-1.2.80.tar.gz"
+
+    maintainers("EbiArnie")
+
+    version("1.2.80", sha256="4c85d736668d82920bfdbdb92ac3d66b7db1108f09581a769dd9160a02def349")
+
+    depends_on("zlib@1.2.3:", type=("build", "link"))
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/kyotocabinet/package.py]: I think this ought to be:
```suggestion
    depends_on("zlib-api@1.2.3:", type=("build", "link"))
```
for better composability with other packages.

File: var/spack/repos/builtin/packages/perl-class-tiny/package.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+# Copyright 2023 EMBL-European Bioinformatics Institute
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/perl-class-tiny/package.py]: This seems to be unrelated to this PR, and a duplicate of #41513

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/kyotocabinet/package.py (hunk 1 above)
Comment (EbiArnie) [reply var/spack/repos/builtin/packages/kyotocabinet/package.py]: OK

File: var/spack/repos/builtin/packages/perl-class-tiny/package.py (hunk 2 above)
Comment (EbiArnie) [reply var/spack/repos/builtin/packages/perl-class-tiny/package.py]: Indeed, sorry about that.

**Review by alalazo — APPROVED** [review]
